wrappers/javascript/packages/anoncreds-nodejs/src/NodeJSAnoncreds.ts

```diff
@@ -617,6 +627,380 @@ export class NodeJSAnoncreds implements Anoncreds {
+        }>
+      >
+    })
+
+    const ret = allocateInt8Buffer()
+
+    this.nativeAnoncreds.anoncreds_verify_w3c_presentation(
+      presentation,
+      presentationRequest,
+      schemas,
+      schemaIds,
+      credentialDefinitions,
+      credentialDefinitionIds,
+      revocationRegistryDefinitions,
+      revocationRegistryDefinitionIds,
+      revocationStatusLists,
+      nonRevokedIntervalOverrideList as unknown as Buffer,
+      ret
+    )
+    this.handleError()
+
+    return Boolean(handleReturnPointer<number>(ret))
+  }
+
+  public credentialToW3C(options: { objectHandle: ObjectHandle; credentialDefinition: ObjectHandle }): ObjectHandle {
+    const { objectHandle, credentialDefinition } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_credential_to_w3c(objectHandle, credentialDefinition, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public credentialFromW3C(options: { objectHandle: ObjectHandle }): ObjectHandle {
+    const { objectHandle } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_credential_from_w3c(objectHandle, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public w3cCredentialAddNonAnoncredsIntegrityProof(options: {
+    objectHandle: ObjectHandle
+    proof: string
+  }): ObjectHandle {
+    const { objectHandle, proof } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_w3c_credential_add_non_anoncreds_integrity_proof(objectHandle, proof, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public w3cCredentialAddNonAnonCredsIntegrityProof(options: {
+    objectHandle: ObjectHandle
+    proof: string
+  }): ObjectHandle {
+    const { objectHandle, proof } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_w3c_credential_add_non_anoncreds_integrity_proof(objectHandle, proof, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public w3cCredentialSetId(options: { objectHandle: ObjectHandle; id: string }): ObjectHandle {
+    const { objectHandle, id } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_w3c_credential_set_id(objectHandle, id, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
+
+  public w3cCredentialSetSubjectId(options: { objectHandle: ObjectHandle; id: string }): ObjectHandle {
+    const { objectHandle, id } = serializeArguments(options)
+
+    const ret = allocatePointer()
+
+    this.nativeAnoncreds.anoncreds_w3c_credential_set_subject_id(objectHandle, id, ret)
+    this.handleError()
+
+    return new ObjectHandle(handleReturnPointer<number>(ret))
+  }
```

Comment:
This feels like it's not really useful in the context of AnonCreds credentials, as you can't use them. Honestly, I'd like to keep it simple and not expose these kind of methods, but just have a simple method like you have with the normal anoncreds encoding (sign based on the attribute map you pass).

That way you can get an anoncreds credential in w3c encoding, if you want to add other contexts, or do special things for using it with other proof types, that should be handled outside the anoncreds library. It has nothing to do with AnonCreds. IMO this library should do the minimal amount of processing it can, and all custom ways should be handled by the frameworks consuming this library.